### DISCUSSION TITLE
Expiration policy not applied to existing cases

### DISCUSSION THREAD

**Discussion by pcinacio** [globaleaks/globaleaks-whistleblowing-software#4212] [General]
### What version of GlobaLeaks are you using?

5.0.11

### What operating system(s) are you seeing the problem on?

Windows, Linux

### Describe the issue

By default, a case expires after 90 days. Setting it to 0 removes the expiration date. For cases that already exist, this change doesn't seem to be applied. For new cases, an expiration date is no longer being defined. Is it supposed to be like this for cases that are already open?
![Screenshot_3](https://github.com/user-attachments/assets/fd73f32d-cbad-431d-9234-aeeacb0dc4f4)
Thanks in advance.

**Comment by karlludwigweise** [comment]
I think this is more like a feature and a result of the architecture of the platform. This way you won't expire cases by accident.

**Comment by evilaliv3** [comment]
Thank you @pcinacio 

I transfromed this ticket into a discussion because for the reasons that i will explain i find it more appropriate.

I support what @karlludwigweise wrote.

Let me explain what has been current evaluations.

The system is very complicated and managing deletion is something to handle with care reducing automation.
We could not make an automatic date of the expiration for reports because this would lead users to loose reports for mistake.
For example if a report by law need to expire tomorrow it would be wrong to extend the deadline automatically; a recipient should choose.
As well if you by mistake change the deadline making it too short, you would risk to delete reports by accident.


Here some suggestions for your system and probably your situation:
If your system was configured with 90 days of data retention policy i invite you to use 365 days or the doubly to have any report expire in one or two years, for a matter of GDPR and similar laws.
If you have a report already received and you want to change its expiration, this can be achieved by recipients that are enabled with the permission to modify the expiration of a report.

Please let me know if this is  clear and if you have any comment

**Comment by pcinacio** [comment]
Thank you both. I understand that, but I'm on the risk to lose a real submission.
The one that has an expiration date on 03/10/2024 01:00 is real, and is triggering email notifications, saying to check it before elimination.
For that reason, i've changed the the retention policy to 0. The test shows no expiration date set for new reports. Great. But what about the other case? I don't want to lose it. How can i set the expiration date to none or extend it?

**Comment by karlludwigweise** [comment]
You can follow this guide to set the expiration date: https://docs.opensource-hinweisgeberportal.de/globaleaks/reports/case-management/set-report-expiration-date

**Comment by pcinacio** [comment]
Something is wrong..
I have a user with the Allow the recipient to edit the report expiration date enabled, but when I open the report, I'm not seeing the option to change the expiration date. That guide has lot more options than me.. 
![Screenshot_1](https://github.com/user-attachments/assets/9ff5fecc-ff1e-4ba1-86ae-07234aefefd1)

![Screenshot_2](https://github.com/user-attachments/assets/70e92b63-4eb7-45ff-aae7-ddcef5d4d74e)

**Comment by karlludwigweise** [comment]
The screenshot from the guide is from an instance with all permissions enabled.

<img width="546" alt="Bildschirmfoto 2024-10-02 um 12 56 43" src="https://github.com/user-attachments/assets/7cd7a37d-22d3-4037-acd6-e14515b7c2a0">

What version are you on? The screenshots are from the latest v4. I don't have a v5 handy to check. It might be a bug in v5 then.

**Comment by pcinacio** [comment]
I have the 5.0.11 running

**Comment by karlludwigweise** [comment]
@evilaliv3 I think this is a bug then.

@pcinacio I don't know, but I think you may be able to change the date in the sqlite database directly, as you only have a few hours left. If you only need the report for documentation purposes, you could also download it to comply with that. Not ideal, I know.

**Comment by pcinacio** [comment]
Leaving mine like your screenshot, I have a few more options, but the one that I need still is missing..

![Screenshot_1](https://github.com/user-attachments/assets/a9cdba27-1569-4041-9cc4-d025aa91f596)

**Comment by karlludwigweise** [comment]
If you have command line access to the machine, I can make you a write-up, how you can fix this rather quickly. Just tested it on one of my instances.

But I don't know about side-effects of just changing this in the sqlite database. @evilaliv3 would know.

**Reply by pcinacio** [reply]
Yes, I have access to the machine. Any time.
I'm watching the database schema. On the table user, I'm believing that the can_postpone_expiration is true, but the button to change it..
Till the end of the day we have time. It's 12:25 here in Portugal. If it can lead to side effects, lets see what @evilaliv3 can share with us. Otherwise, I'll appreciate your guide to force it.

**Reply by karlludwigweise** [reply]
In the table `internaltip` you have the column `expiration_date`. That worked well in my case. I've shut down GlobaLeaks before the change.

If you need more info, let me know.

Fingers crossed.

**Reply by pcinacio** [reply]
Sorry. I was watching this document:
https://docs.globaleaks.org/en/main/developer/DatabaseSchema.html

Where is located the database? It's this ? /var/globaleaks/globaleaks.db

**Reply by karlludwigweise** [reply]
Yes. For non-docker installations the path is `/var/globaleaks/globaleaks.db`.

You can fetch the data from your machine using `scp` via ssh. If you have a ssh config you could use something like this (mac OS):

1. Stop GlobaLeaks.
2. `scp {ssh-config-name-for-host}:/var/globaleaks/globaleaks.db ~/Desktop/globaleaks.db` to copy the database to your local machine.
3. You can open the database with TablePlus on a Mac. Modify `internaltip` table.
4. Upload changes using `scp ~/Desktop/globaleaks.db {ssh-config-name-for-host}:/var/globaleaks/globaleaks.db`
5. Start GlobaLeaks.

You can also use the command line to change the table, but I can't help with that. 😎

**Reply by pcinacio** [reply]
Surprise surprise! What about that? I didn't change anything at all..

![Screenshot_1](https://github.com/user-attachments/assets/b6387c4d-5445-45c8-bb43-3c421935d8d6)

**Reply by karlludwigweise** [reply]
I'd make a local copy before the night just in case.

**Reply by pcinacio** [reply]
Before any change, it was immediatly made.
The report is still there. It was not deleted.

Thank you for your time and attention.
Well, now I have to wait to understand why a datepicker appears in English and not in Portuguese. This, inside the questionnaire, in the question "when did the facts occur?". But, that's another topic/conversation.

**Comment by evilaliv3** [comment]
Thank you @karlludwigweise for your great community support! impressive.

Dear @pcinacio  i'm glad the main problem of expiration has been solved.

We will not check on both the two inconveniences:
- the fact that you did not found the postpone button
- the issue in the date picker

**Comment by evilaliv3** [comment]
@pcinacio : i've fixed the issue on the date picker not opening.

Regarding your impossibility to find the postpone button i'm really not able to see the problem.
Would you please retry with a user with the edit report date priviledge ensuring to act as a recipient in a new tab ? (globaleaks implements some cache that is resetted only opening the browser on a new tab)

**Reply by pcinacio** [reply]
Hi @evilaliv3.
Thank you. I confirm that the picker is opening.
Regarding the postpone button, will be back here to answer soon as possible.
